src/qinfer/hyper_heuristic_optimisers.py

```diff
@@ -0,0 +1,202 @@
+
+
+## FEATURES ###################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+## IMPORTS ####################################################################
+
+import numpy as np
+import random
+from functools import partial
+from qinfer.perf_testing import perf_test_multiple
+from qinfer import distributions
+    
+## CLASSES ####################################################################
+
+__all__ = [
+    'ParticleSwarmOptimiser'
+]
+
+class HyperHeuristicOptimiser(object):
+    '''
+        A generic hyper-heuristic optimiser class that is inherited by the other optimisation functions.
```

Comment:
A short example (such as the ``rosenbrock_fn`` example in your comments on this PR) might go well here. That has the advantage of also providing a doctest check.